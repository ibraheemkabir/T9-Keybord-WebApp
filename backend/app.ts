import express from 'express';
import routes from './src/app-routes/routes';
import bodyParser from 'body-parser';
import { Request, Response,NextFunction,ErrorRequestHandler } from 'express';

const app = express();

const port = 5000;

app.use(bodyParser.json())

app.use(bodyParser.urlencoded({ extended: true }))

app.get('/home',( _ ,res) => {
    res.status(200).json({
        data: 'hello, Welcome to the word converter Api'
    })
});

app.use('/v1', routes)

app.listen(port, () => 
    console.log(`Running on Port ${port}`)
);