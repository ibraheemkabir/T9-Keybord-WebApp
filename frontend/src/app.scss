body {
    background-color: #f6f8fa;
    font-family: 'Dongle', sans-serif;
    color: black;
}

.navbar{
    margin: 1% 2%;
}

.suggestions_container{
    overflow-x: auto;
    overflow-y: none;
    border-bottom-color: #b9b9b92e;
    border-bottom-style: solid;
    border-bottom-width: 1px;
    .suggestion{
        padding: 3px 5px;
        border-right-style: solid;
        border-right-width: 1px;
        border-right-color: #b9b9b92e;
        letter-spacing: 1px;
        cursor: pointer;
    }
   
}

.main_sugestions_container{
    width: 100%;
    padding: 20px 10px;
    text-align: center;

    .header{
        text-align: center;
        font-size: 1.5rem;
    }

    .suggestions_container{
        font-size: 1.5rem;

        .suggestion{
            padding: 3px 5px;
            border-right-style: none;
        }
    }
}

.pagination_footer{
    display: flex;
    justify-content: end;
    margin: 15px auto;
}

.table_info{
    display: flex;
    justify-content: space-between;
   
    .search_input{
        width: 100%;
        width: 35%;
        padding: 0px;
        padding-right: 15px;
        border-radius: 10px;
        background-color: white;
        height: 55px;
        box-shadow: 0px 0px 0px 0px #00003300, 0px 0.5px 0px 0px #00002445, 0px 0px 0px 0px #00001f;
        .MuiInputBase-formControl{
            border-radius: 10px;
        }
    }
}

.table{
    .table_text{
        text-align: center;
        display: flex;
        justify-content: center;
        font-size: 14px;
    }

    .table_item{
        display: flex;
        align-items: center;
        justify-content: center;
    }
}

.list_item{
    margin-bottom: 4px !important;
    box-shadow: none;
    border-radius: 0px !important;
    &:hover{
        transform: scale(1.03);
    }
}

.screen_container{
    min-height: 5rem;
    max-height: 8rem;
    padding: 10px;
    border-top-right-radius: 10px;
    border-top-left-radius: 10px;
    font-size: 25px;
    background-color: #ececec;
    overflow: hidden;

    .inner_screen{
        background-color: #e1e1e1;
        padding: 2px 10px;
        min-height: 6rem;
    }
}

.page_body{
    width: 80%;
    margin: 5% auto;

    .home_body{
        width: 40%;
        margin: 0px auto;
    }

    .list_container{
        border-top-right-radius: 20px;
        border-top-left-radius: 20px;
        border-bottom-left-radius: 0px;
        border-bottom-right-radius: 0px;
        margin-bottom: 4px;
        box-shadow: '0px 0.5px 0.5px -1px rgb(0 0 0 / 20%), 0px 0px 0px 0px rgb(0 0 0 / 14%), 0px 1px 0px 0px rgb(0 0 0 / 12%)';
    
        .list_header{
            font-size: 16px;
            margin-top: 10px;
            letter-spacing:1;
            color:'#a9a9a9';
            max-width:'16.7%';
            font-weight: bold;
        }
        
    }

    .no_data_container{
        text-align: center;
        margin: 50px auto;
        font-size: 25px;
    }
}
