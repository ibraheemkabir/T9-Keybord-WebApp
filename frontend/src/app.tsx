import React from 'react';
import './app.scss';
import {Home} from './pages/Home';
import {AppBar} from './components/AppBar';

export function App(){
    return (
        <div className="page_layout">
            <AppBar/>
            <div className="page_body">
                <Home/>
            </div>
        </div>
    )
}