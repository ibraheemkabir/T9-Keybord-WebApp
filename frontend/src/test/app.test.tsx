import React from 'react';
import {App} from '../app';
import { configure ,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App UI test',()=>{

  test('test app snaphot', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });

  test('test main app elements', () => {
    const component = shallow(<App/>);
    expect(component.find('.page_layout')).toHaveLength(1)
  });
})