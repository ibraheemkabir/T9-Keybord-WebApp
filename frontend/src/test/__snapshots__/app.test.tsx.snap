// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App UI test test app snaphot 1`] = `ShallowWrapper {}`;
